src/par_map.rs

```diff
@@ -0,0 +1,25 @@
+use rayon::iter::{FromParallelIterator, IntoParallelIterator, ParallelIterator};
+
+use crate::Database;
+
+pub fn par_map<Db, D, E, C>(
+    db: &Db,
+    inputs: impl IntoParallelIterator<Item = D>,
+    op: fn(&Db, D) -> E,
+) -> C
+where
+    Db: Database,
+    D: Send,
+    E: Send + Sync,
+    C: FromParallelIterator<E>,
+{
+    dbg!(db.zalsa().views());
+    let fork_db_a: Box<dyn Database> = db.fork_db();
+    let db: &Db = fork_db_a.as_view::<Db>();
+
+    todo!()
+    // inputs
+    //     .into_par_iter()
+    //     .map_with(wrapper, |db, element| op(db, element))
+    //     .collect()
+}
```

Comment:
This is what I imagined:

```rust

pub fn par_map<Db, D, E, C>(
    db: &Db,
    inputs: impl IntoParallelIterator<Item = D>,
    op: fn(&Db, D) -> E,
) -> C
where
    Db: Database,
    D: Send,
    E: Send + Sync,
    C: FromParallelIterator<E>,
{
    let parallel_db = ParallelDb::Ref(db);
    inputs
        .into_par_iter()
        .map_with(parallel_db, |parallel_db, element| {
            let db = parallel_db.as_view::<Db>();
            op(db, element)
        })
        .collect()
}

enum ParallelDb<'db> {
    Ref(&'db dyn Database),
    Fork(Box<dyn Database>),
}

impl Deref for ParallelDb {
    type Target = dyn Database;

    fn deref(&self) -> &Self::Target {
        match self { ... }
    }
}

impl Clone for ParallelDb<'_> {
    fn clone(&self) -> Self {
        ParallelDb::Fork(self.fork())
    }
}
```

Reply:
Oh man, that's way easier. Switched over to this.